### PR TITLE
Add on_leave documentation for require policies

### LINKED ISSUE #14059: Documentation to configure moderated sessions.
Describe the documentation improvements you wish to see.

Looking to improve our documentation for the following option `on_leave`. Recently worked with a customer on configuring moderated sessions, after researching we did not find any documentation on the `on_leave` option. Code signifies when configured the owner can pause or terminate when the owner joins.

More Info:
https://github.com/gravitational/teleport/blob/master/api/types/types.proto#L1907-L1908

Screen Shot 2022-07-01 at 3 30 28 PM

### PR DESCRIPTION
**Pull request by xacrimon**
Fixes #14059

### CODE REVIEW

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 1]
```diff
@@ -178,6 +184,20 @@ filter expression in order to satisfy the policy. The `count` field of a require
 policy is a positive integer value that specifies the minimum amount of users
 this policy requires.
 
+### On leave
+
+The `on_leave` string option in require policies is used to define what happens when a user leaves a session, causing a policy to no longer be satisfied.
```
Comment (zmb3) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:189]:
```suggestion
The `on_leave` string option in require policies is used to define what happens when a moderator leaves a session, causing a policy to no longer be satisfied.
```

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 2]
```diff
@@ -178,6 +184,20 @@ filter expression in order to satisfy the policy. The `count` field of a require
...
+There are two possible actions to take in this scenario:
+1. Terminate the session and disconnect all participants, corresponding to the `"terminate"` value.
+2. Pause the session and stop any input/output streaming until the policy is satisfied again, corresponding to the `"pause"` value.
+
+By default, Teleport treats an empty string in this field as the same as `terminate`.
+That is, the session is terminated instantly and all participants are disconnected.
+
+If all require policies attached to the session owner are set to `"pause"`, the session will instead pause
+but the session will remain open. This discards all input from session participants and buffers output until the session can resume.
+
```
Comment (zmb3) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:200]: Does it buffer or discard output?

Buffering output while a moderator is not present is a security risk, as the user can enter a command that will immediately execute when the moderator returns.

**Review by zmb3 — COMMENTED** [review]
Please also update `docs/pages/includes/role-spec.mdx`, which is meant to be a full role definition showing all available options.

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 3]
```diff
@@ -56,6 +56,12 @@ The following are required options for `require_session_join`:
 |`modes`|`[]`[Participant mode](#participant-modes)|The participant mode that applies to the user joining the Moderated Session under this policy|
 |`count`|Integer|The number of users that need to match the filter expression to satisfy the policy|
 
+The following are options for `require_session_join`:
```
Comment (ptgott) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:59]:
```suggestion
The following fields are optional for `require_session_join`:
```

**Review by ptgott — APPROVED** [review]
Approved with a minor suggestion, adding to **zmb3**'s feedback.

File: docs/pages/access-controls/guides/moderated-sessions.mdx (hunk 2 above)
Comment (xacrimon) [reply docs/pages/access-controls/guides/moderated-sessions.mdx:200]: Did you think of input? I don't see how buffering output (that is, stdout from the shell to the user) could pose a security risk?

Comment (xacrimon) [reply docs/pages/access-controls/guides/moderated-sessions.mdx:200]: I believe the wording does make it clear that all input is indeed discarded.

**Review by zmb3 — APPROVED** [review]
Looks good, thanks!

**Pull request merged**